lua/iswap.lua

```diff
@@ -70,6 +70,148 @@ function M.iswap(config)
   vim.cmd([[silent! call repeat#set("\<Plug>ISwapNormal", -1)]])
 end
 
+
+function M.iswap_cursor_node(config, direction)
+  config = M.evaluate_config(config)
+  local bufnr = vim.api.nvim_get_current_buf()
+  local winid = vim.api.nvim_get_current_win()
+
+	local cursor_node = ts_utils.get_node_at_cursor(winid)
+	local current_row, current_col = cursor_node:start()
+
+
+	-- find outer parent :=  its start() is same as cursor_node:start()
```

Comment:
This works very differently from how I do it currently, if I understand correctly. My queries/ folder supplies specific node types to "bind" to, such as lists and function arguments, which are guaranteed to contain children that can be swapped. This seems a lot more hazardous (but also much more flexible). Should we support both?